src/cluster_asm.c

```diff
@@ -1255,12 +1285,23 @@ void clusterSyncSlotsCommand(client *c) {
             return;
         }
 
+        sds task_id = c->argv[3]->ptr;
         asmTask *task = listLength(asmManager->tasks) == 0 ? NULL :
                             listNodeValue(listFirst(asmManager->tasks));
-        if (task && task->operation == ASM_MIGRATE && task->state == ASM_FAILED &&
-            slotRangeArrayIsEqual(slot_ranges, task->slot_ranges) &&
-            memcmp(task->dest, c->node_id, CLUSTER_NAMELEN) == 0)
-        {
+        if (task && !strcmp(task->id, task_id)) {
+            if (task->operation != ASM_MIGRATE ||
+                !slotRangeArrayIsEqual(slot_ranges, task->slot_ranges) ||
+                memcmp(task->dest, c->node_id, CLUSTER_NAMELEN) != 0)
+            {
+                addReplyError(c, "ID belongs to a different task");
+                zfree(slot_ranges);
+                return;
+            }
+            if (task->state != ASM_FAILED) {
+                addReplyError(c, "Task is already in progress");
+                zfree(slot_ranges);
+                return;
+            }
```

Comment:
how about?
```suggestion
        if (task && !strcmp(task->id, task_id) &&
            task->operation == ASM_MIGRATE && task->state == ASM_FAILED &&
            slotRangeArrayIsEqual(slot_ranges, task->slot_ranges) &&
            memcmp(task->dest, c->node_id, CLUSTER_NAMELEN) == 0)
        {
```
```
 if (task->state != ASM_FAILED) {
                addReplyError(c, "Task is already in progress");
```
a bit duplicated with the following